features/machine/machine.feature

```diff
@@ -357,3 +357,24 @@ Feature: Machine features testing
     And the output should contain:
       | failure-domain.beta.kubernetes.io/zone in [<%= cb.default_zone %>]     |
       | failure-domain.beta.kubernetes.io/region in [<%= cb.default_region %>] |
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-33047
+  @admin
+  @destructive
+  Scenario: Machine API Operator should go degraded if any pod controller is crash looping
+    Given I have an IPI deployment
+    When I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+
+    And I run the steps 20 times or exit when co machine-api is degraded:
+    """
+    Given a pod becomes ready with labels:
+      | api=clusterapi, k8s-app=controller |
+    When admin ensure "<%= pod.name %>" pod is deleted from the "openshift-machine-api" project
+    And the step should succeed
+    """
+    When 180 seconds have passed
```

Comment:
use `And I wait up to 180 seconds for the steps to pass:` to avoid hard code.

Reply:
done